### PR TITLE
1.0

### PR DESCRIPTION
**Pull request by codingknite**

### CODE REVIEW

**Comment by martonlederer** [comment]
Should be good, keep up the good work

**Comment by codingknite** [comment]
Todo 

- [ ] Add Loading State to **New Wallet** Button

File: src/styles/views/Popup/home.module.sass [hunk 1]
```diff
@@ -353,4 +390,41 @@ $sidePaddingTransactions: .7em
...
+.QROverlay
+  position: fixed
+  top: 0
+  left: 0
+  right: 0
+  bottom: 0
+  background-color: rgba(#000, .8)
+  z-index: 1000
+
+  .wrapper
```
Comment (martonlederer) [inline src/styles/views/Popup/home.module.sass:404]: Please use Camel Case

Comment (martonlederer) [inline src/styles/views/Popup/home.module.sass:404]: Please use Camel case in SASS

File: src/styles/views/Welcome/view.module.sass [hunk 2]
```diff
@@ -1,56 +1,107 @@
...
-    width: 30% !important
-    transform: translate(-50%, -50%)
-
-    @media screen and (max-width: variables.$smaller-screens)
-      width: 53% !important
-
-    @media screen and (max-width: variables.$mobile-screens)
-      width: 88% !important
+@import url('https://fonts.googleapis.com/css2?family=Inter&family=Poppins:wght@100;200;300;400;500;600;700;800&display=swap')
+@import url('https://fonts.googleapis.com/css2?family=Inter&family=Poppins:wght@100;200;300;400;500;600;700;800&family=Roboto+Mono:wght@400;500;600&display=swap')
```
Comment (martonlederer) [inline src/styles/views/Welcome/view.module.sass:4]: We don't need the inter font, why do we have that here?
Also Poppins is already added through the ui lib

File: src/styles/views/Welcome/view.module.sass [hunk 3]
```diff
@@ -1,56 +1,107 @@
...
-      margin-bottom: 0
 
-      span
-        text-transform: uppercase
-        font-weight: 600
-        display: inline-block
-
-      @media screen and (max-width: variables.$smaller-screens)
-        font-size: 2.265em
+.startupwelcome
```
Comment (martonlederer) [inline src/styles/views/Welcome/view.module.sass:7]: Camel case pls
`.StartupWelcome`

File: src/views/Popup/routes/Home.tsx [hunk 4]
```diff
@@ -31,6 +40,10 @@ import arweaveLogo from "../../../assets/arweave.png";
 import verto_light_logo from "../../../assets/verto_light.png";
 import verto_dark_logo from "../../../assets/verto_dark.png";
 import styles from "../../../styles/views/Popup/home.module.sass";
+import copy from "copy-to-clipboard";
```
Comment (martonlederer) [inline src/views/Popup/routes/Home.tsx:43]: Just to make sure that everything is organized, let's try to keep the order of imports:
- First imports that don't import the default export of a module (`import { Button }...`)
- Second imports that import the default export of the module (`import copy from...`)
- And the last part should always be the CSS module imports

File: src/views/Popup/routes/Home.tsx [hunk 5]
```diff
@@ -255,30 +270,143 @@ export default function Home() {
     <div className={styles.Home}>
       <WalletManager />
       <div className={styles.Balance}>
-        <h1>
-          {formatBalance(balance()?.arBalance)}
-          <span>AR</span>
-        </h1>
+        <p className={styles.Address}>
+          <button onClick={() => setShowQRCode(true)}>
+            <svg
```
Comment (martonlederer) [inline src/views/Popup/routes/Home.tsx:275]: Please don't use svgs in the code. The icons in the firma are from "iconicicons", the ones that it is missing should be imported from a separate svg file.

File: src/views/Welcome/App.tsx [hunk 6]
```diff
@@ -289,100 +304,155 @@ export default function App() {
     };
   }
 
+  const SetupTemplate = ({
```
Comment (martonlederer) [inline src/views/Welcome/App.tsx:307]: If you have components that are more than 4-5 lines and you reuse them, please create a different file for it in the components directory

**Review by martonlederer — CHANGES_REQUESTED** [review]

File: src/views/Welcome/App.tsx [hunk 7]
```diff
@@ -537,10 +590,11 @@ export default function App() {
...
-          <Input.Password
+          <Input
             {...configPasswordInput.bindings}
+            type="password"
             placeholder="Enter your password to decrypt..."
-            width="100%"
+            style={{ width: "100%" }}
           />
         </Modal.Content>
         <Modal.Action onClick={() => loadConfigModal.setVisible(false)} passive>
```
Comment (martonlederer) [inline src/views/Welcome/App.tsx:596]: Please update the welcome page to use the Verto modals too @codingknite

Comment (codingknite) [reply src/views/Welcome/App.tsx:596]: On it

**Comment by martonlederer** [comment]
Permissions & connect page seems nice @codingknite, just some issues:
- Permissions page checkmarks are not inline
Screenshot 2022-03-29 at PM 5 58 43
- The app name doesn't have the Poppins font
- No need to have a cancel button, we already have the "x" at the top for the allowance modal
- Parts of the UI are almost off the screen for me, try making the fonts just a tiny bit smaller (really just a tiny bit xd)
- The height of the "Select" is not the same as of the input. It probably doesn't have the "small" option in the ui lib, i'll work on that asap

(Maybe you don't have the new ui lib patch, that is why some parts of the ui are showing up differently for you. Pls do a yarn install, thx)

**Comment by codingknite** [comment]
@martonlederer 
> The app name doesn't have the Poppins font

Not sure but it looked to me like the font was already Poppins, maybe adjust the weight?

> Parts of the UI are almost off the screen for me, try making the fonts just a tiny bit smaller (really just a tiny bit xd)

Could you kindly be a bit more specific about these parts so I could make the necessary adjustments

>The height of the "Select" is not the same as of the input. It probably doesn't have the "small" option in the ui lib, i'll work on that asap

No problem

**Comment by martonlederer** [comment]
So the select height is actually not a UI problem, the small style works there.

The parts that are off screen can be found at the screenshot above, but i'll take a screenshot again:
Screenshot 2022-03-31 at PM 2 40 20

**Comment by codingknite** [comment]
Ohh I see, okay cool. Thanks

**Comment by codingknite** [comment]
@martonlederer Input and Select still don't have the same height even with the small option 

Screenshot 2022-04-01 at 01 17 17

**Comment by martonlederer** [comment]
As I said that is not a UI lib problem, could you soend some time debugging that in the extension code?

**Review by t8 — CHANGES_REQUESTED** [review]
Absolutely amazing work here, Joel! Just have a few suggestions before we merge:
<div><div>image</div>
Let's remove the arrow beside the amount to avoid any potential confusion, especially since we already have the explore button below. Can we also remove the swap button for now?</div>
<hr />
<div>
<div>image</div>
Can we make the notifications uniform with the new styles?</div>
<hr />
<div><div>
image
image
</div>
It seems that none of the checkmark buttons change state when clicking them on the connection and permissions pages. Also, can we reduce the spacing between lines on each of the permissions?
</div>
<hr />
<div><div>
image
</div>
Forgive me, but can we think of some type of easter egg to show when a user clicks the hedgehog? This is just begging for an easter egg :D
</div>

cc @martonlederer

**Comment by t8** [comment]
Can't even begin to imagine how @codingknite's commit graph is going to look after this PR is merged 😛

**Pull request closed without merging**